packages/nimble-components/src/table-column/mixins/custom-sort-order.ts

```diff
@@ -0,0 +1,110 @@
+
+// As the returned class is internal to the function, we can't write a signature that uses is directly, so rely on inference
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
+export function mixinCustomSortOrderColumnAPI<
+    TBase extends CustomSortOrderTableColumnConstructor<TColumnValidator>,
+    TColumnValidator extends ColumnValidator<['invalidCustomSortWithGrouping']>
+>(base: TBase) {
+    /**
+     * The Mixin that provides a concrete column with the API to allow sorting
+     * by a field other than the default for that column.
+     */
+    abstract class CustomSortOrderColumn extends base {
+        public sortByFieldName?: string;
+
+        /** @internal */
+        public customSortOrderColumnNotifier?: Notifier;
+
+        /** @internal */
+        public customSortOrderColumnChangeHandler = {
+            handleChange: this.handleCustomSortOrderColumnChange.bind(this)
+        };
+
+        public getResolvedOperandDataRecordFieldName(
+            initialOperandFieldName: string | undefined
+        ): string | undefined {
+            return typeof this.sortByFieldName === 'string'
+                ? this.sortByFieldName
+                : initialOperandFieldName;
+        }
+
+        public getResolvedSortOperation(
+            initialSortOperation: TableColumnSortOperation
+        ): TableColumnSortOperation {
+            return typeof this.sortByFieldName === 'string'
+                ? TableColumnSortOperation.basic
+                : initialSortOperation;
+        }
+
+        /** @internal */
+        public sortByFieldNameChanged(): void {
+            this.handleSortByFieldNameChange();
+            this.updateCustomColumnSortingValidity();
+
+            if (
+                typeof this.sortByFieldName === 'string'
+                && !this.customSortOrderColumnNotifier
+            ) {
+                this.customSortOrderColumnNotifier = Observable.getNotifier(
+                    this.columnInternals
+                );
+                this.customSortOrderColumnNotifier.subscribe(
+                    this.customSortOrderColumnChangeHandler
+                );
+            } else {
+                this.customSortOrderColumnNotifier?.unsubscribe(
+                    this.customSortOrderColumnChangeHandler
+                );
+                this.customSortOrderColumnNotifier = undefined;
+            }
+        }
+
+        public abstract handleSortByFieldNameChange(): void;
+
+        /** @internal */
+        public handleCustomSortOrderColumnChange(
+            _source: unknown,
+            args: unknown
+        ): void {
+            if (args === 'groupingDisabled') {
+                this.updateCustomColumnSortingValidity();
+            }
+        }
```

Comment:
Is this simplified (i.e. avoid the notifier, handler, and all the conditional tracking only if sortBy is configured) if we use groupingDisabledChanged and an optional super invocation?

Something like:

```ts
public groupingDisabledChanged(_source, args) {
    super.groupingDisabledChanged?.(_source, args);
    this.updateCustomColumnSortingValidity();
}
```

Reply:
I don't see a straightforward way to achieve what you are suggesting. `groupingDisabledChanged` is not part of the super class, so I can't invoke it from the mixin, even optionally.

Reply:
@rajsite - wanted to double check that you are fine with me merging this PR without making any changes for this suggestion.

Reply:
Current implementation is valid. I'm guessing it's a typing issue not a runtime issue but not worth spending extra cycles on it. LGTM and thanks for double checking.